common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -518,7 +629,18 @@ def _convert_reference_fields_to_new_namespace(reference):
             else:
                 fields[field_name] = field.read_from(module)
 
-    return store.import_xblock(user_id, dest_course_id, module.location.category, module.location.block_id, fields, runtime)
+    if library:
+        lib = store.get_library(dest_course_id).location
+        return store.create_child(
+            user_id, lib, module.location.category,
+            module.location.block_id, fields,
+            runtime=runtime
+        )
+
+    return store.import_xblock(
+        user_id, dest_course_id, module.location.category,
+        module.location.block_id, fields, runtime
+    )
```

Comment:
Can you be more explicit about which side affects and assumptions? The only thing I see in `import_xblock` that looks like it might not be appropriate is the draft vs published branch selection in split.